doc/api/esm.md

```diff
@@ -1160,25 +1160,26 @@ condition list **must** be passed through to the `defaultResolve` function.
 ```js
 /**
  * @param {string} specifier
- * @param {object} context
- * @param {string} context.parentURL
- * @param {string[]} context.conditions
- * @param {function} defaultResolve
- * @returns {object} response
- * @returns {string} response.url
+ * @param {{
+ *   parentURL:!(string|undefined),
+ *   conditions:!(Array<string>),
```

Comment:
```suggestion
 *   parentURL: !(string|undefined),
 *   conditions: !(Array<string>),
```

Does the spec allow a space here? And what does the `!` mean?

Reply:
Yes, [non-nullable type](https://github.com/google/closure-compiler/wiki/Types-in-the-Closure-Type-System#nullable-vs-optional-parameters-and-properties).

Reply:
I decided to relax Closure Compiler's optimizations to use the default settings so they're gone now. :)

Reply:
> Does the spec allow a space here?

Yes, it does. It would be the equivalent of [the `key-spacing` rule](https://eslint.org/docs/rules/key-spacing), but for JSDoc — stylistic concern.

Is this an actual suggestion or merely a question? Sorry, I was confused.

> And what does the `!` mean?

Non-nullable type, but in this case, it means set of types not including null. 

Refs: https://jsdoc.app/tags-type.html